### ISSUE TITLE
[Bug] iOS requests permission!

### ISSUE THREAD

**Issue by jrahma** [xamarin/Xamarin.Forms#14949]
### Description

The iOS apps seems to be requesting permissions without me knowing that. First it requested calendar and contact permission and archiving my app failed saying I have not set the description in info.plist although I am 1m% sure I am not using calendar nor contact but then I had to add them in info.plist just for archiving to be successful..

After uploading my app successfully to App Store it got rejected because it say I am requesting HealthKit permission when my app is not health app! And yes my app is not health app and I never requested for the HealthKit permission and I cannot see it requested anywhere in my app.

Googling this issue shows it's common issue recently and I am not sure if it's related to recent update.

### Steps to Reproduce

have no idea!

### Expected Behavior
It should not request permission unless I decide to.

### Basic Information

- Version with issue: Latest VS and Latesr XF.

- Last known good version: first time upload

- Platform Target Frameworks: 
iOS

**Comment by jsuarezruiz** [comment]
Could you share a sample or the iOS manifiest file?, what Xamarin.Forms version are you using exactly?

**Comment by jrahma** [comment]
Xamarin Form s is 5.0.0.2244

I can't share a sample because it happened on this file but I can share it privately.

**Comment by jenscski** [comment]
Do you have any third party libraries installed?

We had issues with a library that was causing a similar issue for us.

**Comment by angelru** [comment]
You can always appeal or add those permissions. Sometimes apple is like this ...

**Comment by jrahma** [comment]
> Do you have any third party libraries installed?
> 
> We had issues with a library that was causing a similar issue for us.

No, I don't. They are all the same regular Nuget Packages I am using for all apps [Attached]


[Archive.zip](https://github.com/xamarin/Xamarin.Forms/files/7655183/Archive.zip)

**Comment by jrahma** [comment]
and here is my Build options

<img width="729" alt="Screen Shot 2021-12-05 at 8 00 28 AM" src="https://user-images.githubusercontent.com/6236241/144732910-92c457d2-d22c-4012-8357-782a02acb419.png">

**Comment by jfversluis** [comment]
> Googling this issue shows it's common issue recently

What sources do you have on this?

I am fairly sure Forms does not add any permissions whatsoever. Either it's something you (mistakenly) enabled yourself of some other library that you maybe added. Also, all permissions should be declared in the `info.plist` file and the `AndroidManifest.xml` file so you should easily see what is in there.

**Comment by jrahma** [comment]
> > Googling this issue shows it's common issue recently
> 
> What sources do you have on this?
> 
> I am fairly sure Forms does not add any permissions whatsoever. Either it's something you (mistakenly) enabled yourself of some other library that you maybe added. Also, all permissions should be declared in the `info.plist` file and the `AndroidManifest.xml` file so you should easily see what is in there.

It does request event if you don’t add it to the info.plist

For example, in my case, I never added the HealthKit and I double-checked my info.plist and 100% I don’t have the HealthKit.. However, my iOS app got rejected twice for the HealthKit permission request.

Following this article

[https://docs.microsoft.com/en-us/answers/questions/286148/rejected-app-due-to-healthkit-healthkit-is-not-ref.html](https://docs.microsoft.com/en-us/answers/questions/286148/rejected-app-due-to-healthkit-healthkit-is-not-ref.html
)

I tried the workaround of changing the Linker setting and uploaded the app and now I am waiting for apple's apporoval.

Below is the screenshot of my info.plist where you will also notice that I am requesting a permission for Contacts and Calendar. This was a (force request) which I had to do because my app got rejected and the reason was that Ia m requesting permission for both although I did not so I had to add the permission inside the info.plist just for the approval process

<img width="290" alt="Screen Shot 2021-12-06 at 7 34 47 PM" src="https://user-images.githubusercontent.com/6236241/144875272-d8d3a5ea-8df4-44e7-8ffd-c439d7a03300.png">

<img width="274" alt="Screen Shot 2021-12-06 at 7 34 57 PM" src="https://user-images.githubusercontent.com/6236241/144875340-9ea16359-cb00-448a-a1df-4a53e7dc5a43.png">

**Comment by jfversluis** [comment]
I still don't see a "common issue" by Googling this. There is one report, that you linked, the follow-up on that is here: https://github.com/xamarin/xamarin-macios/issues/10743 where the author stops responding.

Also, this issue seems to belong to the macios repo and not here anyway. Could you please post the exact message that you received from Apple?

@dalexsoto does this mean anything to you and do you have the powers to move this?

**Comment by dalexsoto** [comment]
The problem is the linker behavior 

![image](https://user-images.githubusercontent.com/204671/144878130-f0ced5be-519b-496a-acea-f63d56ec2969.png)

It is set to don't link which means that the app is getting all of the symbols referenced in his App which includes all of the frameworks like HealthKit etc. This is why Apple is flagging the app. The solution is to enable the linker setting to do either just SDK (which will remove only not used SDK bits) or full linker which will try to link everything.

Docs: https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/linker?tabs=macos

**Comment by jrahma** [comment]
> I still don't see a "common issue" by Googling this. There is one report, that you linked, the follow-up on that is here: [xamarin/xamarin-macios#10743](https://github.com/xamarin/xamarin-macios/issues/10743) where the author stops responding.
> 
> Also, this issue seems to belong to the macios repo and not here anyway. Could you please post the exact message that you received from Apple?
> 
> @dalexsoto does this mean anything to you and do you have the powers to move this?

This is the exact message. and as expected, I did not change any permission because I don't have the HealthKit anywhere in my app but by just changing the Linker setting my app was approved today.

![apple_message](https://user-images.githubusercontent.com/6236241/145087045-a4b721f2-a7ce-455d-b64b-5a03d657e56f.png)

**Comment by jrahma** [comment]
> TL;DR;: not a bug, the problem is the linker behavior
> 
> ![image](https://user-images.githubusercontent.com/204671/144878130-f0ced5be-519b-496a-acea-f63d56ec2969.png)
> 
> It is set to don't link which means that the app is getting all of the symbols referenced in his App which includes all of the frameworks like HealthKit etc. This is why Apple is flagging the app. The solution is to enable the linker setting to do either just SDK (which will remove only not used SDK bits) or full linker which will try to link everything.
> 
> Docs: https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/linker?tabs=macos

but this setting does not make sense. If Do not Link creates such issue then why it's available in the list?!

Visual Studio should know what's used in my app and it should add everything just because I set it to Do not Link.

**Comment by jfversluis** [comment]
> If Do not Link creates such issue then why it's available in the list?!

You can read up on the different settings and their use here: https://docs.microsoft.com/xamarin/ios/deploy-test/linker

I'm going to assume things are clear and this is indeed not a bug. Let me know if that's not the case. Thanks!

**Comment by dalexsoto** [comment]
@jrahma Just answering your question about that setting, it exists because certain applications can be distributed outside the AppStore and also some kind of applications like the ones that allow you to code on device like Continuous can take advantage of not linking everything. 

Also some debug scenarios are faster if we do not link at all for example on simulators so there are a lot of applications for such kind of setting.

**Comment by jrahma** [comment]
@jfversluis it will be great if you consider this as enhancement to avoid confusion and rejections of app which will help the developers and make things clear to them and save their time

**Comment by jfversluis** [comment]
> @jfversluis it will be great if you consider this as enhancement to avoid confusion and rejections of app which will help the developers and make things clear to them and save their time

Even if we were to do that, the Xamarin.Forms repository is not the right place for this feature request.

Personally I don't really see much opportunity for an enhancement here. This setting has been here for a long, long time and I have never seen a big number of people with issues like this one. By default this setting is configured properly whenever you select the Release configuration. 

Additionally I think we may assume a certain level of experience/knowledge about the tools that are being worked with. If not, we are always here to help. I don't think everything is something that needs a technical solution, sometimes it's just knowledge :)
